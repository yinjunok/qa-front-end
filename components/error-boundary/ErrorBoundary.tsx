import * as React from 'react';

interface IErrorBoundaryState {
  hasError: boolean;
}

interface IErrorBoundaryProps {
  errorCom: React.ReactNode;
}

class ErrorBoundary extends React.Component<IErrorBoundaryProps, IErrorBoundaryState> {
  static getDerivedStateFromError() {
    return {
      hasError: true,
    }
  }

  static defaultProps = {
    errorCom: null,
  }

  state = {
    hasError: false,
  }

  render() {
    if (this.state.hasError) {
      return this.props.errorCom;
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
