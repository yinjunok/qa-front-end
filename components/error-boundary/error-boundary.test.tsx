import renderer from 'react-test-renderer';
import ErrorBoundary from './ErrorBoundary';

function TestCom({ error = true }) {
  if (error) {
    throw new Error('has error');
  }

  return <h1>no error</h1>
}

describe('test snapshots', () => {
  it('has error', () => {
    const com = renderer.create(
      <ErrorBoundary>
        <TestCom error={true} />
      </ErrorBoundary>
    );

    expect(com.toJSON()).toMatchSnapshot();
  });

  it('normal', () => {
    const com = renderer.create(
      <ErrorBoundary>
        <TestCom error={false} />
      </ErrorBoundary>
    );

    expect(com.toJSON()).toMatchSnapshot();
  });

  it('custom error component', () => {
    const com = renderer.create(
      <ErrorBoundary errorCom={<h1>custom error component</h1>}>
        <TestCom error={true} />
      </ErrorBoundary>
    );

    expect(com.toJSON()).toMatchSnapshot();
  });
});
