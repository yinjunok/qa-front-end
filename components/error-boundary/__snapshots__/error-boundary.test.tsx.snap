// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshots custom error component 1`] = `
<h1>
  custom error component
</h1>
`;

exports[`test snapshots has error 1`] = `null`;

exports[`test snapshots normal 1`] = `
<h1>
  no error
</h1>
`;
